src/__tests__/indexing.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * @jest-environment ./build/index.js
+ */
+import { jest } from '@jest/globals'
+import { CeramicApi, Page, StreamState, StreamUtils, TestUtils } from '@ceramicnetwork/common'
+import { ModelInstanceDocument } from '@ceramicnetwork/stream-model-instance'
+import { StreamID } from '@ceramicnetwork/streamid'
+import { config } from 'node-config-ts'
+import { createDid } from '../utils.js'
+import { DID } from 'dids'
+
+const TEST_MODEL = StreamID.fromString(config.jest.models[0])
+
+declare global {
+  const ceramic: CeramicApi
+  const ceramicClient: CeramicApi
+}
+
+const extractStreamStates = (page: Page<StreamState | null>): Array<StreamState> => {
+  if (page.edges.find(edge => edge.node === null)) {
+    throw new Error(
+      'null stream state found. This may indicate a problem with data persistence of your state store, which can result in data loss.'
+    )
+  }
+
+  return page.edges.map(edge => edge.node) as Array<StreamState>
+}
+
+const extractDocuments = (
+  ceramic: CeramicApi,
+  page: Page<StreamState | null>
+): Array<ModelInstanceDocument> => {
+  return extractStreamStates(page).map(state =>
+    ceramic.buildStreamFromState<ModelInstanceDocument>(state)
+  )
+}
+
+describe('indexing', () => {
+  describe('Using existing model', () => {
+    jest.setTimeout(1000 * 60)
+    const originalDid = ceramic.did as DID
+
+    const singleNodeTestCases: any[] = []
+    const twoNodesTestCases: any[] = []
+
+    if (config.jest.services.ceramic.indexingEnabled) {
+      singleNodeTestCases.push(['ceramic', ceramic])
+    }
+
+    if (config.jest.services.ceramicClient.indexingEnabled) {
+      singleNodeTestCases.push(['ceramicClient', ceramicClient])
+    }
+
+    if (
+      config.jest.services.ceramic.indexingEnabled &&
+      config.jest.services.ceramicClient.indexingEnabled
+    ) {
+      twoNodesTestCases.push(['creating: ceramic, loading: ceramicClient', ceramic, ceramicClient])
+      twoNodesTestCases.push(['creating: ceramicClient, loading: ceramic', ceramicClient, ceramic])
+    }
+
+    afterEach(() => {
+      ceramic.did = originalDid
+      ceramicClient.did = originalDid
+    })
+
+    if (singleNodeTestCases.length > 0) {
+      test.each(singleNodeTestCases)(
+        'Can create and query on same node -- %s',
+        async (_, ceramicInstance: CeramicApi) => {
+          const doc1 = await ModelInstanceDocument.create(
+            ceramicInstance,
+            { data: Math.floor(Math.random() * 1000) },
+            { model: TEST_MODEL },
+            { anchor: false }
+          )
+
+          await TestUtils.delay(5 * 1000)
+
+          const resultsAfterCreate = await ceramicInstance.index
+            .query({ model: TEST_MODEL, last: 10 })
+            .then(resultObj => extractDocuments(ceramicInstance, resultObj))
+          expect(resultsAfterCreate.length).toBeGreaterThanOrEqual(1)
+
+          const retrievedCreatedDoc1 = resultsAfterCreate.at(-1) as ModelInstanceDocument
```

Comment:
Does this mean that we have newer docs showing up last in the result set?  That seems unintuitive to me.  @PaulLeCam, should we switch the default sort order to be that newer docs come first?

Reply:
My understanding of the spec, especially when it comes to [edge order](https://relay.dev/graphql/connections.htm#sec-Edge-order) is that the connection slicing parameters shouldn't affect ordering in any way.
We can decide that our ordering is in reverse order by default, so the `last` would come first but that means the `first` would come last, which is even more confusion IMO.
Basically, GraphQL connections only care than ordering is consistent, however we manage this consistency and ordering choice is up to us.

Reply:
right exactly.  So my question is, when requesting `first: 10`, should you get the 10 most recently created docs, or the 10 oldest documents?

Reply:
10 oldest IMO, as the default ordering is chronological